test/e2e/space_autocompletion_test.go

```diff
@@ -166,6 +169,78 @@ func TestAutomaticClusterAssignment(t *testing.T) {
+			WithTargetClusterRoles([]string{cluster.RoleLabel("workspace")})) // request that specific cluster role
+
+		// then
+		VerifyResourcesProvisionedForSpace(t, awaitilities, space1.Name, wait.UntilSpaceHasStatusTargetCluster(memberAwait2.ClusterName))
+	})
+
+	t.Run("set cluster-role label only on member2 cluster but mark it as full so that no cluster will be available", func(t *testing.T) {
+		// given
+		// only member1 as room for spaces
+		hostAwait.UpdateToolchainConfig(t, testconfig.CapacityThresholds().MaxNumberOfSpaces(
+			testconfig.PerMemberCluster(memberAwait1.ClusterName, 500), // member1 has more room
+			testconfig.PerMemberCluster(memberAwait2.ClusterName, -1),  // member2 is full
+		))
+		// let's add a custom cluster-role for member2
+		memberCluster2, found, err := hostAwait.GetToolchainCluster(t, memberAwait2.Type, memberAwait2.Namespace, nil)
+		require.NoError(t, err)
+		require.True(t, found)
+		_, err = hostAwait.UpdateToolchainCluster(t, memberCluster2.Name, func(tc *toolchainv1alpha1.ToolchainCluster) {
+			tc.Labels[cluster.RoleLabel("workspace")] = "" // add a new cluster-role label, the value is blank since only key matters.
+		})
+		require.NoError(t, err)
+
+		// when
+		space1, _ := CreateSpaceWithBinding(t, awaitilities, mur, WithName("space-clusterole-tenant-pending"),
+			WithTargetClusterRoles([]string{cluster.RoleLabel("workspace")})) // request that specific cluster role
+
+		// then
+		waitUntilSpaceIsPendingCluster(t, hostAwait, space1.Name)
+	})
+
+	t.Run("provision space on preferred cluster even if it doesn't match the specified cluster-role", func(t *testing.T) {
+		// given
+		hostAwait.UpdateToolchainConfig(t, testconfig.CapacityThresholds().MaxNumberOfSpaces(
+			testconfig.PerMemberCluster(memberAwait1.ClusterName, 500), // member1 has more room
+			testconfig.PerMemberCluster(memberAwait2.ClusterName, 500), // member2 has more room as well
+		))
+		// let's add a custom cluster-role for member2
+		memberCluster2, found, err := hostAwait.GetToolchainCluster(t, memberAwait2.Type, memberAwait2.Namespace, nil)
+		require.NoError(t, err)
+		require.True(t, found)
+		_, err = hostAwait.UpdateToolchainCluster(t, memberCluster2.Name, func(tc *toolchainv1alpha1.ToolchainCluster) {
+			tc.Labels[cluster.RoleLabel("workspace")] = "" // add a new cluster-role label, the value is blank since only key matters.
+		})
+		require.NoError(t, err)
+
+		// when
+		space1, _ := CreateSpaceWithBinding(t, awaitilities, mur, WithName("space-preferred-tenant"),
+			WithTargetClusterRoles([]string{cluster.RoleLabel("workspace")}), WithTargetCluster(memberAwait1.ClusterName)) // request that specific cluster role and preferred cluster which will have priority on the roles
+
+		// then
+		// space should end up on preferred cluster even if it doesn't have the specified cluster roles
+		VerifyResourcesProvisionedForSpace(t, awaitilities, space1.Name, wait.UntilSpaceHasStatusTargetCluster(memberAwait1.ClusterName))
```

Comment:
Let's not confuse two terms:
1. the required target cluster (specified in the spec)
2. the preferred target cluster (taken from UserSignup.Status)
```suggestion
	t.Run("provision space on the required cluster even if it doesn't match the specified cluster-role", func(t *testing.T) {
		// given
		hostAwait.UpdateToolchainConfig(t, testconfig.CapacityThresholds().MaxNumberOfSpaces(
			testconfig.PerMemberCluster(memberAwait1.ClusterName, 500), // member1 has more room
			testconfig.PerMemberCluster(memberAwait2.ClusterName, 500), // member2 has more room as well
		))
		// let's add a custom cluster-role for member2
		memberCluster2, found, err := hostAwait.GetToolchainCluster(t, memberAwait2.Type, memberAwait2.Namespace, nil)
		require.NoError(t, err)
		require.True(t, found)
		_, err = hostAwait.UpdateToolchainCluster(t, memberCluster2.Name, func(tc *toolchainv1alpha1.ToolchainCluster) {
			tc.Labels[cluster.RoleLabel("workspace")] = "" // add a new cluster-role label, the value is blank since only key matters.
		})
		require.NoError(t, err)

		// when
		space1, _ := CreateSpaceWithBinding(t, awaitilities, mur, WithName("space-required-tenant"),
			WithTargetClusterRoles([]string{cluster.RoleLabel("workspace")}), WithTargetCluster(memberAwait1.ClusterName)) // request that specific cluster role and required cluster which will have priority on the roles

		// then
		// space should end up on the required cluster even if it doesn't have the specified cluster roles
		VerifyResourcesProvisionedForSpace(t, awaitilities, space1.Name, wait.UntilSpaceHasStatusTargetCluster(memberAwait1.ClusterName))
```

Reply:
Thanks for pointing this out! I got confused between required and preferred, changed it in [552c9de](https://github.com/codeready-toolchain/toolchain-e2e/pull/656/commits/552c9de9298aecfd678d30c5fe86917526b3e699)